lib/adal.js

```diff
@@ -175,18 +185,62 @@ var AuthenticationContext = (function () {
         this._saveItem(this.CONSTANTS.STORAGE.NONCE_IDTOKEN, this._idTokenNonce);
         this._saveItem(this.CONSTANTS.STORAGE.ERROR, '');
         this._saveItem(this.CONSTANTS.STORAGE.ERROR_DESCRIPTION, '');
-
-
         var urlNavigate = this._getNavigateUrl('id_token', null) + '&nonce=' + encodeURIComponent(this._idTokenNonce);
-        this.frameCallInProgress = false;
         this._loginInProgress = true;
+        if (this.popUp) {
+            this._loginPopup(urlNavigate);
+            return;
+        }
         if (this.config.displayCall) {
             // User defined way of handling the navigation
             this.config.displayCall(urlNavigate);
         } else {
             this.promptUser(urlNavigate);
         }
-        // callback from redirected page will receive fragment. It needs to call oauth2Callback
+    };
+
+    AuthenticationContext.prototype._openPopup = function (urlNavigate, title, popUpWidth, popUpHeight) {
+        // adding winLeft and winTop to account for dual monitor
+        // using screenLeft and screenTop for IE8 and earlier
+        try {
+            var winLeft = window.screenLeft ? window.screenLeft : window.screenX;
+            var winTop = window.screenTop ? window.screenTop : window.screenY;
+            //window.innerWidth displays browser window's height and width excluding toolbars
+            //using document.documentElement.clientWidth for IE8 and earlier
+            var width = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+            var height = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
+            var left = ((width / 2) - (popUpWidth / 2)) + winLeft;
+            var top = ((height / 2) - (popUpHeight / 2)) + winTop;
+
+            var popupWindow = window.open(urlNavigate, title, 'width=' + popUpWidth + ', height=' + popUpHeight + ', top=' + top + ', left=' + left);
+            if (popupWindow.focus) {
+                popupWindow.focus();
+            }
+            return popupWindow;
+        } catch (e) {
+            this.info(e.message);
+            this.loginInProgress = false;
+            return null;
+        }
+    }
+
+    AuthenticationContext.prototype._loginPopup = function (urlNavigate) {
+        var popupWindow = this._openPopup(urlNavigate, "login", 483, 600);
```

Comment:
since we return null from _openPopup and also, in some browsers window.open won't work, we should check if 'popupWindow' is null, if so, log an error message and set the LOGIN_ERROR constant.